aten/src/ATen/native/SoftMax.cpp

```diff
@@ -465,23 +482,62 @@ Tensor log_softmax(const Tensor& self, Dimname dim, optional<ScalarType> dtype)
   return at::log_softmax(self, dimname_to_position(self, dim), dtype);
 }
 
-Tensor masked_softmax_cpu(const Tensor& input, const Tensor& mask) {
+Tensor masked_softmax_cpu(const Tensor& input, int64_t dim, const Tensor& mask) {
```

Comment:
Could you please make this dim optional, and set the by-default to be input.dim()-1?

Reply:
Done!